community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -226,6 +227,41 @@ else if ( seconds > 0 && nanos < 0 )
         this.nanos = (int) nanos;
     }
 
+    public int compareTo( DurationValue other )
+    {
+        return Comparator.comparingLong(DurationValue::averageLengthInNanos)
+                .thenComparingLong( DurationValue::getMonths )
+                .thenComparingLong( DurationValue::getDays )
+                .thenComparingLong( DurationValue::getSeconds )
+                .thenComparingInt( DurationValue::getNanos ).compare( this, other );
+    }
+
+    private long averageLengthInNanos()
+    {
+        return this.nanos + this.seconds * NANOS_PER_SECOND + this.days * NANOS_PER_SECOND * SECONDS_PER_DAY +
+                this.months * NANOS_PER_SECOND * (long) (SECONDS_PER_DAY * AVERAGE_DAYS_PER_MONTH);
+    }
+
+    public long getMonths()
+    {
+        return months;
+    }
+
+    public long getDays()
+    {
+        return days;
+    }
+
+    public long getSeconds()
+    {
+        return seconds;
+    }
+
+    public int getNanos()
+    {
+        return nanos;
+    }
```

Comment:
If these methods are for the comparator only, there is no need to make them public, in fact they could be inlined into the comparator as a lambda accessing the field directly (e.g. `d -> d.months`).